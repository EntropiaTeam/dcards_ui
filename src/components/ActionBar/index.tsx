import React, { FC, ReactNode } from 'react';
import {
  Box, makeStyles, Grid, Theme
} from '@material-ui/core';
import MainButton from '../UI/MainButton';

const useStyles = makeStyles((theme: Theme) => ({
  footerContainer: {
    display: 'flex',
    alignItems: 'flex-end',
    padding: 27,
    [theme.breakpoints.only('xs')]: {
      padding: '0 13px 27px 13px'
    },
    width: '100%',
    boxSizing: 'border-box'
  }
}));

interface Props {
  mainButtonContent: ReactNode;
  mainButtonCallback(): void;
  className?: string;
  mainButtonDisabled?: boolean;
  tabIndex?: number;
  navLinks?: ReactNode;
}

const ActionBar: FC<Props> = ({
  mainButtonContent,
  mainButtonCallback,
  mainButtonDisabled,
  className = '',
  tabIndex = 0,
  navLinks
}) => {
  const classes = useStyles();
  return (
    <Box className={`${classes.footerContainer} ${className}`}>
      <Grid container>
        <Grid item md={4} lg={4} xl={4} />
        <Grid item xs={8} sm={8} md={4} lg={4} xl={4} justify="space-around" container>
          {navLinks}
        </Grid>
        <Grid item xs={4} sm={4} md={4} lg={4} xl={4} justify="flex-end" container>
          <MainButton
            disabled={mainButtonDisabled}
            onClick={mainButtonCallback}
            tabIndex={tabIndex}
          >
            {mainButtonContent}
          </MainButton>
        </Grid>
      </Grid>
    </Box>
  );
};

export default ActionBar;
