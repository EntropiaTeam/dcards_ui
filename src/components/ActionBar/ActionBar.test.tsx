import React from 'react';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';
import ActionBar from '.';

describe('<ActionBar />', () => {
  const props = {
    navLinks: [1, 2, 3, 4, 5],
    mainButtonCallback: jest.fn()
  };

  const ActionBarComponent = renderer.create(<ActionBar {...props} />).toJSON();
  const wrapper = mount(<ActionBar {...props} />);

  it('should render ActionBar component correctly', () => {
    expect(ActionBarComponent).toMatchSnapshot();
  });

  it('should render ComponentNavLinks with given navlinks', () => {
    const ComponentNavLinksText = wrapper.find('.MuiGrid-justify-xs-space-around').text();
    expect(ComponentNavLinksText).toBe(props.navLinks.join(''));
  });

  it('should render ActionBar component with proper navLinks prop', () => {
    expect((wrapper).prop('navLinks')).toEqual(props.navLinks);
  });
});
