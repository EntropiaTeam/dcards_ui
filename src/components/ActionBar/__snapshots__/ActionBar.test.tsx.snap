// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ActionBar /> should render ActionBar component correctly 1`] = `
<div
  className="MuiBox-root MuiBox-root-2 makeStyles-footerContainer-1 "
>
  <div
    className="MuiGrid-root MuiGrid-container"
  >
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-md-4 MuiGrid-grid-lg-4 MuiGrid-grid-xl-4"
    />
    <div
      className="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-justify-xs-space-around MuiGrid-grid-xs-8 MuiGrid-grid-sm-8 MuiGrid-grid-md-4 MuiGrid-grid-lg-4 MuiGrid-grid-xl-4"
    >
      1
      2
      3
      4
      5
    </div>
    <div
      className="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-justify-xs-flex-end MuiGrid-grid-xs-4 MuiGrid-grid-sm-4 MuiGrid-grid-md-4 MuiGrid-grid-lg-4 MuiGrid-grid-xl-4"
    >
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-button-3 MuiButton-containedPrimary makeStyles-primary-4 MuiButton-disableElevation"
        disabled={false}
        onBlur={[Function]}
        onClick={[MockFunction]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiButton-label"
        />
      </button>
    </div>
  </div>
</div>
`;
